### DISCUSSION TITLE
Yolov8x on openvino: hammering my cpu and high inference speed

### DISCUSSION THREAD

**Discussion by doctorttt** [blakeblackshear/frigate#6811] [General]
@aeozyalcin I've been following some of your posts where you helped a number of folks to get yolo to work with openvino, so I thought directing my question to you might be appropriate? For the frigate team members and other contributors, please feel free to chime in.

I used aeozyalcin's colab to produce a yolov8x.xml file and got openvino to work with yolov8x in frigate. However, inference speed was very bad at 500+ms for three cameras with detection on and my cpu (12900k) got hammered.  I'm running the project's official docker image - v0.12.1 stable

I went back to the colab and produced a yolov8n xml file; loaded it to frigate; and inference speed immediately improved to 25-30ms for the same three cameras with detection on. 

I'm new to this space. Is yolov8x too much for a cpu to handle in general or I did something wrong?

**Comment by aeozyalcin** [comment]
There are multiple factors that impact the inference speed:

- Size & resolution of the model you are using. Yolov8n at 320x320 will be an order of magnitude faster than Yolov8m at 640x640, at the expense of accuracy. Yolov8x is a HUGE model, so I am not surprised at all that it would take 500+ms inference time. That model is really more suitable to run on server grade NVidia GPUs, and not for real time inference. I'd recommend sticking with Yolov8m and smaller models at <640x640, you'll be happy with the performance.
- Hardware. I looked up the 12900k, and confirmed that it still comes with integrated graphics, which is good for OpenVino performance. Some Intel Desktop CPUs are CPU-only, and don't come with integrated graphics. As long as OpenVino is configured to make use of the integrated GPU in your Intel SoC, you are good to go.
- Camera resolution. For inference, I don't recommend feeding your 4K camera stream to the inference engine. Make use of the substreams for better inference performance.
- Many more....

**Reply by NickM-27** [reply]
To be clear camera resolution will not affect inference speed (as it is shown in frigate system page). The same size region is always sent to the model for inference and the inference speed in frigate does not include resizing the region to the model size.

**Comment by doctorttt** [comment]
Thank you guys, much appreciated. 

I tweaked some settings and confirmed that openvino is now using my Intel igpu (i could see a detector line under intel_gpu_top in Unraid). In addition, I tested Yolov8n, Yolov8s, and Yolov8m all at 416 x 416. 

- Yolov8n with detection on three cameras: ~10 ms inference time
- Yolov8s: ~20 ms inference time
- Yolov8m: around ~100 ms inference time. 

In general, as the model gets bigger, the inference time takes longer. 

I'm going to stick with v8n or v8s. Agreed that an Nvidia GPU would be more suitable. I will close this discussion now. Thanks again.
